BeVolt/App/src/CLI.c

```diff
@@ -0,0 +1,723 @@
+			toggleLED(CAN);
+			break;
+		case EXTRA_HASH:
+			toggleLED(EXTRA);
+			break;
+		default:
+			printf("Invalid LED command\n\r");
+			break;
+	}
+}
+
+/** CLI_CAN
+ * Interacts with CAN by
+ * reading and writing to bus
+ */
+void CLI_CAN(void) {
+	uint8_t readData;
+	uint8_t writeData[8];
+	for(uint8_t i = 0; i < 7; i+= 2) {
+		char tempData[2] = {hexString[i], hexString[i+1]};
+		writeData[i/2] = strtol(tempData, NULL, 16);
+	}
+	switch(hashTokens[1]) {
+		case READ: 
+			while(CAN1_Read(&readData)){
+				printf("%d\n\r ", readData);
+			}
+		break;
+		case WRITE: 
+			switch(hashTokens[2]) {
+				case CAN_TRIP:
+					CAN1_Write(CAN_ID_BPS_TRIP, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case CLEAR:
+					CAN1_Write(CAN_ID_BPS_ALL_CLEAR, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case OFF:
+					CAN1_Write(CAN_ID_BPS_OFF, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case CURRENT:
+					CAN1_Write(CAN_ID_CURRENT_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case VOLTAGE:
+					CAN1_Write(CAN_ID_TOTAL_VOLTAGE_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case TEMPERATURE:
+					CAN1_Write(CAN_ID_AVG_TEMPERATURE_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case CHARGE:
+					CAN1_Write(CAN_ID_SOC_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case WATCHDOG:
+					CAN1_Write(CAN_ID_WDOG_TRIGGERED, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case ERROR:
+					CAN1_Write(CAN_ID_ERROR, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				default:
+					printf("Invalid ID\n\r");
+					break;
+			}
+			break;
+		default:
+			printf("Invalid CAN command\n\r");
+			break;
+	}
+}
+	
+// TODO: Confirm status of Display;
+//		 if operational, display proper information (specifics found on Drive)
+/** CLI_Display
+ * Interacts with the display
+ */
+void CLI_Display(void) {}
+
+/** CLI_Watchdog
+ * Shows whether watchdog was tripped
+ */
+void CLI_Watchdog(void) {
+	if(hashTokens[1] == NULL) {
+		printf("Safety Status: ");
+		if (WDTimer_DidSystemReset() == SAFE){
+			printf("SAFE\n\r");
+		} else if (WDTimer_DidSystemReset() == DANGER){
+			printf("DANGER\n\r");
+		}
+	}
+	uint8_t errorAddrArray[2];
+	EEPROM_ReadMultipleBytes(EEPROM_WATCHDOG_PTR_LOC, 2, errorAddrArray);
+	uint16_t errorAddr = (errorAddrArray[0] << 2) + errorAddrArray[1];
+	switch(hashTokens[1]) {
+		case ERROR:
+			printf("Most recent Watchdog error: %d", EEPROM_ReadByte(errorAddr-1));
+			break;
+		default:
+			printf("Invalid Watchdog command");
+			break;
+	}
+}
+
+/** CLI_EEPROM
+ * Interacts with EEPROM
+ * by reading and writing to the EEPROM
+ */
+void CLI_EEPROM(void) {
+	if(hashTokens[1] == NULL) {
+		EEPROM_SerialPrintData();
+		return;
+	}
+	uint8_t errorAddrArray[2];
+	uint16_t errorAddr = 0;

```

Comment:
Two similarly names variables could use at least some comments explaining the difference between them.

Is errorAddr even and addr? or is it just an error?

Reply:
Added explanation. The way EEPROM gives the values (from what I understand), I need to get the address of the error from the pointer first, then access the error itself.